### PR TITLE
Audio: Part 1

### PR DESCRIPTION
**Pull request by ethteck**
This includes a rough decomp of many audio functions. Many structs are not fully figured out, and surely there are some things that need to be merged and some types that need to change. I found two cases where unions were seemingly necessary, but these may prove to not be needed eventually. 

I also did some data disassembly / migration as part of this branch, and there are a few other assorted function decomps.

### CODE REVIEW

File: src/code_2e230_len_2190.c [hunk 1]
```diff
@@ -1,48 +1,442 @@
...
+
+
+    temp4 = *temp_s4;
+    temp_v0_2 = alHeapAlloc(alHeap, 1, 0x8000);
+    temp4->unk_5C = &temp_v0_2[0];
+    temp4->unk_60 = &temp_v0_2[0x1400];
+    temp4->unk_64 = &temp_v0_2[0x1C00];
+    temp4->unk_68 = &temp_v0_2[0x1400];
+
+    for (i = 0; i < 1; i++) {
```
Comment (nanaian) [inline src/code_2e230_len_2190.c:44]: love this

**Review by nanaian — APPROVED** [review]
lgtm! awesome work 💯

File: src/code_2e230_len_2190.c (hunk 1 above)
Comment (ethteck) [reply src/code_2e230_len_2190.c:44]: lol

File: src/code_25f00_len_940.c [hunk 2]
```diff
@@ -1,18 +1,112 @@
...
+            break;
+        case 1:
+            temp_a0_2 = D_800A0F50;
+            temp_s0 = temp_a0_2 - (temp_a0_2 / 20) * arg1;
+
+            if (temp_s0 < 0) {
+                temp_s0 = 0;
+            }
+
+            temp_s0 = (temp_s0 * temp_s0) >> 0xF;
```
Comment (pixel-stuck) [inline src/code_25f00_len_940.c:98]: this seems suspect. The ``>> 0xF`` is probably really ``/ 0x8000``

File: src/code_26840_len_20d0.c [hunk 3]
```diff
@@ -1,18 +1,199 @@
...
+    arg0->unk_84[4] = 0x20;
+    arg0->unk_84[5] = 0x30;
+    arg0->unk_84[6] = 0x40;
+    arg0->unk_84[7] = 0x50;
+
+    func_8004B9E4(arg0, 0);
+}
+
+INCLUDE_ASM(void, "code_26840_len_20d0", func_8004B62C, UnkAl6CC* arg0);
+// void func_8004B62C(UnkAl6CC *arg0) {
```
Comment (pixel-stuck) [inline src/code_26840_len_20d0.c:87]: this should probably be removed

File: src/code_28910_len_5090.c [hunk 4]
```diff
@@ -54,19 +70,115 @@ INCLUDE_ASM(s32, "code_28910_len_5090", func_8004DCB8);
...
 
-INCLUDE_ASM(s32, "code_28910_len_5090", func_8004E3A4);
+void func_8004E3A4(UnkAlA9C* arg0) {
+    arg0->unk_2C.unk_0A--;
 
-INCLUDE_ASM(s32, "code_28910_len_5090", func_8004E444);
+    if (arg0->unk_2C.unk_0A != 0) {
+        arg0->unk_2C.unk_00.s32 += arg0->unk_2C.unk_04;
+    } else {
+        arg0->unk_2C.unk_00.s32 = arg0->unk_2C.unk_08 << 0x10;
```
Comment (pixel-stuck) [inline src/code_28910_len_5090.c:164]: ``<< 16`` here, preferably

File: src/code_28910_len_5090.c [hunk 5]
```diff
@@ -84,63 +196,248 @@ INCLUDE_ASM(s32, "code_28910_len_5090", func_8004EC04);
...
+    arg0->unk_B8 = 0;
+    arg0->unk_B4 = 0;
+}
+
+s32 func_8004FC08(UnkAlA9C* arg0, u32 arg1) {
+    u32 t32;
+    u16 t16;
+    u32 ret;
+
+    t32 = t16 = arg0->unk_20A;
```
Comment (pixel-stuck) [inline src/code_28910_len_5090.c:217]: could this potentially just be achieved with a cast?

File: src/code_28910_len_5090.c [hunk 6]
```diff
@@ -84,63 +196,248 @@ INCLUDE_ASM(s32, "code_28910_len_5090", func_8004EC04);
...
+
+    ret = arg1;
+    ret *= arg0->unk_D0;
```
Comment (pixel-stuck) [inline src/code_28910_len_5090.c:220]: can this potentially be rewritten as ``ret = arg1 * arg0->unk_D0;`` ?

**Review by pixel-stuck — CHANGES_REQUESTED** [review]

File: Makefile [hunk 7]
```diff
@@ -128,7 +128,7 @@ setup: clean submodules tools split $(LD_SCRIPT)
 # tools/star-rod submodule intentionally omitted
 submodules:
 	git submodule init tools/n64splat
-	git submodule update --init --recursive
+	git submodule update --recursive
```
[the remark this answers is not in the document]
Comment (nanaian) [reply Makefile:131]: star-rod is optional

File: src/code_28910_len_5090.c (hunk 6 above)
Comment (ethteck) [reply src/code_28910_len_5090.c:220]: Unfortunately no, that screws with reg alloc.

Comment (pixel-stuck) [reply src/code_28910_len_5090.c:220]: rip

File: src/code_25f00_len_940.c (hunk 2 above)
Comment (ethteck) [reply src/code_25f00_len_940.c:98]: That adds in some other instructions unfortunately

Comment (pixel-stuck) [reply src/code_25f00_len_940.c:98]: huh. Well, if we're leaving this, then ``>> 15`` would be preferrable

**Pull request merged**